src/client/game.cpp

```diff
@@ -3358,6 +3358,9 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud)
 	if (g_touchcontrols) {
 		auto mode = selected_def.touch_interaction.getMode(pointed.type);
 		g_touchcontrols->applyContextControls(mode);
+		// update again so that TOSERVER_INTERACT packets have the correct controls set
+		client->getEnv().getLocalPlayer()->control.dig = isKeyDown(KeyType::DIG);
+		client->getEnv().getLocalPlayer()->control.place = isKeyDown(KeyType::PLACE);
```

Comment:
1.
```suggestion
		player->control.dig = isKeyDown(KeyType::DIG);
		player->control.place = isKeyDown(KeyType::PLACE);
```

2. Wouldn't this be better in `applyContextControls()`? Having it here looks somewhat messy.

Reply:
> Wouldn't this be better in `applyContextControls()`? Having it here looks somewhat messy.

Not really, `TouchControls` doesn't and shouldn't know about `LocalPlayer` or `PlayerControl`.